### ISSUE TITLE
Role and User Gating

### ISSUE THREAD

**Issue by Behold3th** [BanklessDAO/bountybot#3]
Conversation started here:
https://github.com/Behold3th/discord-bots/issues/4

**Comment by Behold3th** [comment]
**TL; DR: **
**Do we accept users and roles input in the subcommand menu? If yes, how many of each?
Do we accept users and roles input in the DM flow? If yes, how should they be notated?**

An initial implementation of role and user gating for bounties is live in my 'Local Only Bot' 

There are some tough questions for this feature from a UX perspective in discord due to the discord platforms UX for slash and subcommands. The current implementation only takes in one role and one user.

In the auto-complete sub command menu, there is no way to easily take in an array of `Mentionables`. `Mentionables` are a huge add to the customer experience. When a user starts to type in a `Mentionable`(the character '@', followed by the first character of the user or role), discord pulls up a list of fitting users and roles - thus eliminating headaches from spelling errors and double checking the name and discriminant of a user.

Due to discord's slash command registration architecture, each subcommand in a subcommand group (the subcommand group here being `bounty create`) needs a unique name. Thus, we can only take in a static amount of roles and users. To add more than one user and role in the sub command menu context, I would define additional users and roles through the namespace convention: user_1, user_2, user_3.... and role_1, role_2, role_3 ... 

So our first option is on the number of users and roles we accept in the slash command menu. I think acceptable choices are 0, 1, and 3.

The next question is how we utilize the DM flow to help with this feature. Our DM flow already consists of asking for:

- Description
- Completion Criteria
- Date

Logically, we would add two steps to this flow:

- Users
- Roles

However, adding users and roles in the DM flow will be difficult for the UX. Since the interaction is now outside the context of a guild, it's no longer possible to mention the needed users and roles. So, the user would be entering in plain text for both these requests, without the benefit of the `Mentionable` discord UX. One fork in the road here is whether to have the user input ids for the users and roles (they would have to enable developer mode, then navigate to a user message in the central discord pane or online status in the left discord pane, right click and select `Copy ID`) or names for roles and names and discriminants for users. Names and roles means the headache of spelling mistakes and double checking. Ids means the headache of pasting unreadable values into an array (anticipate a user losing track of which ones they've already entered ... yikes). 

As well, if we accept the input of Users and Roles in the DM flow, should it be additive to users and roles entered in the sub command menu? Or should it be definitive (I.e. overwriting previously entered values)? 

**My recommendation in the order of posed questions:** 
1 user and 1 role in the sub command menu
DM flow accepts users and roles
In DM flow, users notated by name and discriminant. Before storing in db, check for validity
In DM flow, roles notated by name. Before storing in db, check for validity
